### PR TITLE
Improve error message in case too much nesting

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes https://github.com/rinja-rs/rinja/issues/28.

### CODE REVIEW

File: rinja_parser/src/lib.rs [hunk 1]
```diff
@@ -711,10 +711,10 @@ pub(crate) struct Level(u8);
 impl Level {
     fn nest(self, i: &str) -> ParseResult<'_, Level> {
         if self.0 >= Self::MAX_DEPTH {
-            return Err(ErrorContext::from_err(nom::Err::Failure(error_position!(
+            return Err(nom::Err::Failure(ErrorContext::new(
+                "reached nested depth limit",
```
Comment (Kijewski) [inline rinja_parser/src/lib.rs:715]: Maybe something like `"your template code is too deeply nested, or last expression is too complex"`?

Comment (GuillaumeGomez) [reply rinja_parser/src/lib.rs:715]: Fine by me!

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by Kijewski — APPROVED** [review]
Yep, much easier to understand. :)

**Pull request merged**